Content.Shared/Chemistry/Components/Solution.cs

```diff
@@ -320,98 +482,71 @@ public FixedPoint2 GetTotalPrototypeQuantity(params string[] prototypes)
         }
 
         /// <summary>
-        ///     Adds a given quantity of a reagent directly into the solution.
-        /// </summary>
-        /// <param name="prototype">The prototype ID of the reagent to add.</param>
-        /// <param name="quantity">The quantity in milli-units.</param>
-        public void AddReagent(string prototype, FixedPoint2 quantity, bool dirtyHeatCap = true)
-            => AddReagent(new ReagentId(prototype, null), quantity, dirtyHeatCap);
-
-        /// <summary>
-        ///     Adds a given quantity of a reagent directly into the solution.
-        /// </summary>
+        ///     Adds a reagent to a solution.
         /// <param name="id">The reagent to add.</param>
-        /// <param name="quantity">The quantity in milli-units.</param>
-        public void AddReagent(ReagentId id, FixedPoint2 quantity, bool dirtyHeatCap = true)
+        /// <param name="quantity">How much to add</param>
+        /// <param name="protoMan">Prototype manager, optional.</param>
+        /// <param name="temperature">Temperature of the reagent, optional.</param>
+        /// </summary>
+        public void AddReagent(ReagentId id, FixedPoint2 quantity, IPrototypeManager? protoMan = null, float? temperature = null)
```

Comment:
To make the AddReagent code less spaghetti-y and to have less overloaded AddReagent functions. Also i just had to add a PrototypeManager arg on every function because it is needed to get the lowest and highest phase transition temperature reagents.

Reply:
Resolving the prototype manager and updating all the properties is expensive and will significantly slow down adding reagents, it should defer the processing as much as possible. PrototypeManager should also be a mandatory argument wherever possible to avoid people not passing it in